mm2src/komodefi_cli/src/scenarios/download_helper.rs

```diff
@@ -0,0 +1,103 @@
+struct Releases {
+    assets: Vec<Asset>,
+}
+
+/// Let komodefi_cli download and extract release binary for use.
+pub(crate) async fn download_binary_and_extract_to_bin_folder() -> Result<()> {
+    // Create a reusable reqwest client with common headers
+    let client = reqwest::Client::builder()
+        .user_agent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36")
+        .default_headers({
+            let mut headers = reqwest::header::HeaderMap::new();
+            headers.insert(reqwest::header::ACCEPT, "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7".parse().unwrap());
+            headers
+        })
+        .build()?;
+    let response = client.get(BINARY_HOST_URL).send().await?;
+    let releases: Vec<Releases> = response.json().await?;
+
+    // Extract the download URL for the latest release
+    if let Some(download_url) = releases.get(0).and_then(|release| {
+        release
+            .assets
+            .iter()
+            .find(|asset| asset.name.contains(DOWNLOAD_NAME))
+            .map(|asset| &asset.browser_download_url)
+    }) {
+        // Download the ZIP file
+        let zip_data = client.get(download_url).send().await?.bytes().await?;
+        // Create directories if they don't exist
+        let bin_dir = std::env::current_dir()?.join("bin");
+        if !bin_dir.exists() {
+            std::fs::create_dir_all(&bin_dir)?;
+        }
+        // Save the ZIP file
+        let zip_path = bin_dir.join("downloaded_file.zip");
+        let mut zip_file = File::create(&zip_path)?;
+        zip_file.write_all(&zip_data)?;
+        // Extract only mm2 binary file from the folder
+        extract_file_from_zip(&zip_path, BINARY_NAME)?;
+        info!("Binary downloaded and extracted to the bin folder!");
+        Ok(())
+    } else {
+        error!("No matching release found");
+        Err(anyhow::anyhow!("No matching release found"))
+    }
+}
+
+/// Extract binary file from zip file
+fn extract_file_from_zip(zip_path: &std::path::Path, file_name: &str) -> Result<(), anyhow::Error> {
+    let file = File::open(zip_path)?;
+    let reader = std::io::BufReader::new(file);
+
+    let mut archive = ZipArchive::new(reader)?;
+
+    // Find the index of the desired file in the archive
+    let file_index = archive
+        .file_names()
+        .position(|name| name == file_name)
+        .ok_or_else(|| anyhow::anyhow!("File not found in the ZIP archive"))?;
+
+    let mut file = archive.by_index(file_index)?;
+
+    // Create directories if they don't exist
+    let bin_dir = std::env::current_dir()?.join("bin");
+    if !bin_dir.exists() {
+        std::fs::create_dir_all(&bin_dir)?;
+    }
+
+    let file_path = bin_dir.join(file_name);
+    let mut extracted_file = File::create(&file_path)?;
+
+    std::io::copy(&mut file, &mut extracted_file)?;
```

Comment:
Checking the binary by something like `mm2 --version` would be nice before copying it.

Reply:
what about doing this check after the extraction/copy?

Reply:
It can't be done before the extraction. But I think it should be before copying it. Do we have any reason for copying the broken binary?

Reply:
You mean you are now using extraction path directly without moving the binary files? If so, it depends where you download-extract the build file (e.g., if you are doing it in common dirs like /tmp or /downloads, it can be removed automatically or manually by the user).

Reply:
uhmm it goes like this,

- I download the binary zip to the bin folder
- extract the zip file content to the bin folder 
- get the binary version and log
- and finally delete the zip file

Reply:
where do you download/extract it ?

Reply:
`/bin` folder same place where `komodefi_cli` binary is saved

Reply:
It's not /bin, and it's not `komodefi_cli/bin` either. You are reading current directory, which can be literally anywhere. An example, when I call `komodefi_cli` without being in the same directory as `komodefi_cli` (like `./path/komodefi_cli mm2 download` or `komodefi_cli mm2 download` from the the bin path exported in PATH, it will download and extract to where I execute the command). To achieve what you aiming to do you should use https://doc.rust-lang.org/std/env/fn.current_exe.html. But this isn't good either because  users will need to run this cli with root permissions when they have the binary when it's installed under `/bin`. It's much better to create dir at home like `.komodefi_cli` and keep everything there (like configuration and the mm2 binary), and make that path configurable via environment variable. e.g., `KOMODEFI_CLI_ROOT=/var/lib/komodefi-cli komodefi-cli` should use `/var/lib/komodefi-cli` as downloading bin and creating configurations, and if user didn't set `KOMODEFI_CLI_ROOT`, use the `HOME_PATH/.komodefi_cli` by default.

Reply:
@onur-ozkan this enhancement has been done and I also updated cli docs for this enhancement.